src/navigation/reroute.js

```diff
@@ -69,8 +70,14 @@ export function reroute(pendingPromises = [], eventArguments) {
     return loadApps();
   }
 
-  function cancelNavigation() {
-    navigationIsCanceled = true;
+  function cancelNavigation(promise) {
+    if (!navigationIsCanceled) {
```

Comment:
This if statement isn't doing anything, since `navigationIsCanceled` is never set to true.

```suggestion
```